draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1441,7 +1490,48 @@ Steps:
 8. return u
 ~~~
 
-# Deterministic Mappings  {#mappings}
+## Alternative hash\_to\_base functions {#hashtobase-alt}
+
+The hash\_to\_base function is suitable for use with a wide range of hash functions,
```

Comment:
A possible clarification:

```
The hash\_to\_base function is suitable for use with a wide range of hash functions,
including SHA-2 {{FIPS180-4}}, SHA-3 {{FIPS202}}, BLAKE2 {{!RFC7963}}, and others.
In some cases, however, implementors may wish to replace the HKDF-based function
defined in this section with one that uses a different underlying pseudorandom generator.
(The function in this section uses HKDF-Expand as a variable-length pseudorandom function
since it accepts an additional context parameter as input. In practice, however, any pseudorandom
generator would suffice provided context strings are supported.) Examples of different
pseudorandom functions include extendable-output functions like cSHAKE {{SP.800-185}}
or BLAKE2X {{BLAKE2X}}.
```

I think this aligns the PRF vs PRG distinction that Hugo makes in his paper, and I hope avoids confusion. (Some readers might observe that HKDF-Expand is a PRF whereas we now refer to it as PRG.)

Reply:
You're right, I was sloppy. We should probably use PRF throughout. Let me try again... :)

Reply:
OK, I've pushed some clarifications.

One major change is that the section does not recommend any particular primitives, because it seems like not a great idea to recommend something unless we're quite sure it will work.

(For example, thinking about it more, I think KMAC is a better option than cSHAKE---so I changed the one remaining example in the Suites section.)